### PR TITLE
Remove all uses of code in Equality compare/Equality compare and branch categories

### PR DESCRIPTION
**Pull request by wbh123456**
Remove all uses of IL Opcodes in `Equality compare` and `Equality compare and branch` categories.
  
Did not delete:
- `intCompareNarrower(node, s, TR::ifsucmpeq, TR::ifscmpeq, TR::ifbcmpeq);`
- `longCompareNarrower(node, s, TR::ificmpne, TR::ifsucmpne, TR::ifscmpne, TR::ifbcmpne)`
- `sucmpeqEvaluator `
- `sucmpneSimplifier`

Issue:#2657
Signed-off-by: Bohao(Aaron) Wang <aaronwang0407@gmail.com>

### CODE REVIEW

File: compiler/optimizer/SwitchAnalyzer.cpp [hunk 1]
```diff
@@ -627,7 +627,7 @@ TR::Block *TR::SwitchAnalyzer::peelOffTheHottestValue(TR_LinkHead<SwitchInfo> *c
 
       TR::Block *newBlock = NULL;
 
-      cmpOp = _isInt64 ? (_signed ? TR::iflcmpeq : TR::iflucmpeq) : (_signed ? TR::ificmpeq : TR::ifiucmpeq);
+      cmpOp = _isInt64 ? TR::iflcmpeq : TR::ificmpeq;
```
Comment (0xdaryl) [inline compiler/optimizer/SwitchAnalyzer.cpp]: Please do the SwitchAnalyzer changes as part of #3983 .

**Comment by wbh123456** [comment]
Revert changes to `SwitchAnalyzer.cpp` back in the new force push.

**Comment by wbh123456** [comment]
Did not remove:
1. `TR::ifsucmpeq` in `intCompareNarrower(node, s, TR::ifsucmpeq, TR::ifscmpeq, TR::ifbcmpeq);`

    - Reason: `TR::ifsucmpeq` is used along with other opcodes. Not sure how to remove `TR::ifsucmpeq`.

2. `TR::ifsucmpne` in `longCompareNarrower(node, s, TR::ificmpne, TR::ifsucmpne, TR::ifscmpne, TR::ifbcmpne)`
     - Reason: `TR::ifsucmpne` is used along with other opcodes. Not sure how to remove `TR::ifsucmpne`.
3. `sucmpeqEvaluator`
    - Reason: `TR::sucmpeq` has been removed, but not sure if we can remove `sucmpeqEvaluator` .
4. `sucmpneSimplifier`
    - Resaon: `TR::sucmpne` has been removed, but not sure if we can remove `sucmpneSimplifier` .

**Comment by 0xdaryl** [comment]
Thanks for the changes.  Rather than remove the SwitchAnalyzer changes in one commit, and then add them back in a separate commit, can you simply not remove them in the first place?  It means you will need to edit one of your commits when you rebase.  I just think it is cleaner that way.

**Comment by wbh123456** [comment]
> Thanks for the changes. Rather than remove the SwitchAnalyzer changes in one commit, and then add them back in a separate commit, can you simply not remove them in the first place? It means you will need to edit one of your commits when you rebase. I just think it is cleaner that way.

Fixed in the new force push.

**Comment by 0xdaryl** [comment]
@genie-omr build all

**Pull request merged**